web/angular-wallet/src/app/setup/account/account.service.ts

```diff
@@ -202,29 +202,49 @@ export class AccountService {
   */
   public synchronizeAccount(account: Account): Promise<Account> {
     return new Promise(async (resolve, reject) => {
-      try {
-
-        const remoteAccount = await this.getAccount(account.account);
-        account.name = remoteAccount.name;
-        account.description = remoteAccount.description;
-        account.assetBalances = remoteAccount.assetBalances;
-        account.unconfirmedAssetBalances = remoteAccount.unconfirmedAssetBalances;
-        account.balanceNQT = remoteAccount.balanceNQT;
-        account.unconfirmedBalanceNQT = remoteAccount.unconfirmedBalanceNQT;
-
-        const transactionList = await this.getAccountTransactions(account.account);
-        account.transactions = transactionList.transactions;
-
-        const unconfirmedTransactionsResponse = await this.getUnconfirmedTransactions(account.account);
-        account.transactions = unconfirmedTransactionsResponse.unconfirmedTransactions
-          .concat(account.transactions);
-
-      } catch (e) {
-        console.log(e);
-      }
+      // todo: parallelize
+      await this.syncAccountDetails(account);
+      await this.syncAccountTransactions(account);
+      await this.syncAccountUnconfirmedTransactions(account);
```

Comment:
These are mutating and could be made in parallel, but I can do that later.

Reply:
:+1: Promise.all() is your friend, if you need to return a promise then...